src/uu/date/src/date.rs

```diff
@@ -223,10 +223,17 @@ pub fn uumain(args: impl uucore::Args) -> UResult<()> {
                 Box::new(iter)
             }
             DateSource::File(ref path) => {
-                file = File::open(path).unwrap();
-                let lines = BufReader::new(file).lines();
-                let iter = lines.filter_map(Result::ok).map(parse_date);
-                Box::new(iter)
+                match File::open(path) {
+                    Ok(file) => {
+                        let lines = BufReader::new(file).lines();
+                        let iter = lines.filter_map(Result::ok).map(parse_date);
+                        Box::new(iter)
+                    }
+                    Err(_) => {
+                        eprintln!("Error: File not found at path {:?}", path);
```

Comment:
please use the same wording as GNU
```
$ LANG=C date -f file-non-existing
date: file-non-existing: No such file or directory
```

Reply:
Also, I think the syntax should be like:
https://github.com/uutils/coreutils/blob/a75cfd96555f44198b5410f65ec501f89cc6ef30/src/uu/sum/src/sum.rs#L88-L93

and the error code should be 1

Reply:
My bad! Fixed :)

Reply:
this isn't resolved ;) see my second comment (using USimpleError)

Reply:
Aye! Sorry!

Would this work? I'm sorry for bugging you, just new to GNU contributions!

```rust
Err(err) => {
                        return Err(USimpleError::new(
                            1,
                            format!("failed to open {}: {}", path.display(), err),
                        ));
                    }
```

Reply:
please try
but the error message isn't the one I asked you to update ;)

Reply:
Error code now: `date: aze: No such file or directory`
Tests: All passed (added one for testing valid and then one for testing invalid files)
Clippy: No warnings

Reply:
Error code now: `date: aze: No such file or directory`
Tests: All passed (added one for testing valid and then one for testing invalid files)
Clippy: No warnings